src/Sobol.jl

```diff
@@ -1,67 +1,93 @@
 module Sobol
 
-import NLopt.libnlopt
-import Base: ndims, length, skip, next, start, done, show, writemime
+import Base: ndims, skip, next, start, done, show, writemime
 export SobolSeq, ScaledSobolSeq, next!
 
-type SobolSeq{N}
-    p::Ptr{Void} # nlopt_sobol pointer
-    function SobolSeq()
-        (N < 1 || N > 1111) && error("invalid Sobol dimension")
-        p = ccall((:nlopt_sobol_create,libnlopt), Ptr{Void}, (Cuint,), N)
-        p == C_NULL && error("error initializing Sobol sequence")
-        s = new(p)
-        finalizer(s, destroy)
-        s
-    end
+include("soboldata.jl") #loads `sobol_a` and `sobol_minit`
+
+type SobolSeq
+    sdim::Int #dimension of sequence being generated
+    m::Array{UInt,2} #array of size (sdim, WORD_SIZE)
+    x::Array{UInt,1} #previous x = x_n, array of length sdim
+    b::Array{UInt,1} #position of fixed point in x[i] is after bit b[i]
+    n::UInt #number of x's generated so far
 end
-SobolSeq(N::Integer) = SobolSeq{N}()
 
-destroy(s::SobolSeq) = 
-    ccall((:nlopt_sobol_destroy,libnlopt), Void, (Ptr{Void},), s.p)
+ndims(s::SobolSeq) = s.sdim
 
-ndims{N}(s::SobolSeq{N}) = N
+function SobolSeq(N::Integer)
+    (N < 1 || N > 1111) && error("invalid Sobol dimension")
 
-function show{N}(io::IO, s::SobolSeq{N})
-    print(io, "$N-dimensional Sobol sequence on [0,1]^$N")
-end
-function writemime{N}(io::IO, ::MIME"text/html", s::SobolSeq{N})
-    print(io, "$N-dimensional Sobol sequence on [0,1]<sup>$N</sup>")
-end
+    m = ones(UInt, (N, WORD_SIZE))
 
-function skip!{N}(s::SobolSeq{N}, n::Integer, x::Vector{Float64})
-    (n < 0 || length(x) < N) && throw(BoundsError())
-    ccall((:nlopt_sobol_skip,libnlopt), Void, (Ptr{Void}, Cuint, Ptr{Float64}),
-          s.p, n, x)
-    x
-end
+    if N == 1 #special case
+        return SobolSeq(N,m,UInt[0],UInt[0],zero(UInt))
+    end
 
-function skip{N}(s::SobolSeq{N}, n::Integer) 
-    skip!(s, n, Array(Float64, N))
-    nothing
+    for i = 2:N
+        a = sobol_a[i-1]
+        d = floor(UInt, log2(a)) #degree of poly
+
+        #set initial values of m from table
+        # TODO transpose sobol_minit for faster column access?
+        m[i, 1:Int(d)] = sobol_minit[i-1, 1:Int(d)]
```

Comment:
Just make `d = floor(Int, log2(a))` and then you don't need the subsequent `Int(d)` calls.